app/models/user.rb

```diff
@@ -6,7 +6,68 @@ class User < ApplicationRecord
 
   validates :name, presence: true, length: { maximum: 20 }
 
+  # rubocop:disable Layout/LineLength:
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :recieved_friendship, class_name: 'Friendship', foreign_key: 'reciever_user_id', inverse_of: :reciever_user, dependent: :destroy
+  has_many :requested_friendship, class_name: 'Friendship', foreign_key: 'requester_user_id', inverse_of: :requester_user, dependent: :destroy
+  # rubocop:enable Layout/LineLength:
+  def friends
+    friends = []
+    Friendship.all.each do |request|
+      if request.status == true
+        if request.requester_user_id == id
+          friends << User.find(request.reciever_user_id)
+        elsif request.reciever_user_id == id
+          friends << User.find(request.requester_user_id)
+        end
+      end
+    end
+    friends
+  end
```

Comment:
Using enumerable methods here is not a good practice. The best way to go there is to add a scope that renders all the users where the request.status == true. Check [this article](https://www.ducktypelabs.com/using-scope-with-associations/) to know more.